spark/v3.2/spark/src/test/java/org/apache/iceberg/spark/source/TestSparkReaderDeletes.java

```diff
@@ -139,7 +193,21 @@ protected void dropTable(String name) {
   }
 
   @Override
-  public StructLikeSet rowSet(String name, Table table, String... columns) {
+  protected boolean countDeletes() {
```

Comment:
Instead of the apis here, could we just modify "extractDeleteCount()" 

to instead be something like "
```java 
long lastMetricCount(String metricName) {
// Get executionID of last run thing
// Find metricName
// return value for MetricName
}
```

This way we can avoid some state in the job?

Reply:
I removed the `deleteCount` field and the `setDeleteCount(long)` method, and the `deleteCount()` method now simply calls a `lastExecutedMetricValue(...)` method along the lines you suggest.